dts/arm/st/stm32f0.dtsi

```diff
@@ -113,6 +113,16 @@
 			status = "disabled";
 			label = "SPI_1";
 		};
+
+		spi2: spi@40003800 {
```

Comment:
spi2 is not present on all f0 SoCs so it shouldn't be in this place. 
I'd propse adding an stm32f030Xc.dtsi and a stm32f030X8.dtsi file and adding the node there.

Reply:
I thought changing the status to "ok" in relevant board would be sufficient. (one that support only spi1, or both etc...), since the base address seems to be the same (well, as you noticed below: if f09x or f07x has spi2 support but with different base address then splitting in relevant files will be necessary)

Reply:
there's been some discussion on this in #5883, eg this: https://github.com/zephyrproject-rtos/zephyr/pull/5883#issuecomment-361517351

Reply:
I did what you proposed yes